File: 240421/2개씩 그룹짓기/group-of-pairs.py
N = int(input())
data = list(map(int,input().split()))
data.sort()

max_val = data[0]
for i in range(N):
    val = data[i] + data[2*N-1-i]
    max_val = max(val,max_val)
print(max_val)

File: 240603/2개씩 그룹짓기/group-of-pairs.java
import java.util.*;
import java.util.stream.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        List<Integer> inputs = Arrays.stream(br.readLine().split(" "))
            .map(Integer::parseInt)
            .collect(Collectors.toList());

        inputs.sort(Comparator.naturalOrder());

        int answer = Integer.MIN_VALUE;
        for (int i=0; i<inputs.size()/2; i++) {
            int tempSum = inputs.get(i) + inputs.get(inputs.size() - 1 - i);
            answer = Math.max(tempSum, answer);
        }
        System.out.println(answer);
    }
}